src/HydraAuction/Delegate/Interface.hs

```diff
@@ -0,0 +1,35 @@
+module HydraAuction.Delegate.Interface (
+  FrontendKind (..),
+  KnownFrontendRequest (..),
+  DelegateError (..),
+  DelegateResponse (..),
+  FrontendRequest (..),
+) where
+
+-- Prelude imports
+import Prelude
+
+-- Cardano imports
+import Cardano.Api.UTxO (UTxO)
+
+-- HydraAuction imports
+import HydraAuction.Types (AuctionTerms (..), Natural)
+
+data FrontendKind = Seller | Bidder
```

Comment:
I would remove this distinction as well. It is out of scope to handle authentication between the bidders and the delegates, we can just trust what the CLI says for the demo.

Reply:
Yes this is the datatype which defines who is client trusting CLI. Not trusting is postponed to another issue #168 

All logic code required for trusting-client-case is already in this PR.

Reply:
We already do that, my code not requires to actually authenticate client, only store it role.

Anyway, we have a discussion with @GeorgeFlerovsky and found out that I was wrong, and it seems like there is no reasons to authenticate client at all or limit delegate to having one client. So I gonna change state to only store `AuctionTerms`.